### ISSUE TITLE
Modify unittests 

### ISSUE THREAD

**Issue by andreas-bok-sociomantic** [sociomantic/libdrizzle-redux#7]
Currently `unittests` depends on environment variables to set up connection to `mysql` db

```
tests/unit/binlog.cc:73 - main()

drizzle_st *con= drizzle_create(getenv("MYSQL_SERVER"),
    getenv("MYSQL_PORT") ? atoi("MYSQL_PORT") : DRIZZLE_DEFAULT_TCP_PORT,
    getenv("MYSQL_USER"),
    getenv("MYSQL_PASSWORD"),
    getenv("MYSQL_SCHEMA"), 0);
```

Only `MYSQL_PORT` has a fallback value if it is not defined. Hence if `MYSQL_USER, MYSQL_PASSWORD, MYSQL_SCHEMA` defined the tests will fail. 

Possible solutions
- Keep the dependency on environment variables. But info about `env var` dependency
- Move mysql config to a settings file in the `test/` folder
- Add default values for connection parameters in test source code. 

What are your thoughts, @leandro-lucarella-sociomantic , @ben-palmer-sociomantic

**Comment by ben-palmer-sociomantic** [comment]
For `MYSQL_SERVER` it makes sense to me to use `127.0.0.1` but for user, password, and schema I don't see obvious default values. Perhaps a settings file in the test folder would be the best way to go.

**Comment by leandro-lucarella-sociomantic** [comment]
Yeah, I think defaults in here will make very little sense. I don't mind keeping the env vars. Using a config file could be also nice, but bear in mind that file shouldn't be part of the repo, as it's local configuration. The config file can be simply a trivial shell script defining the variables:

``` sh
MYSQL_SERVER=blah
MYSQL_PORT=1
# etc.
```

then `. config.sh; run-tests`.

**Comment by leandro-lucarella-sociomantic** [comment]
In any case, it should be well documented, and ideally the test should check all the env vars are defined and give a sensible error (asking to set the variables) if it's not.

Another option, if it's an option, would be to pass the mysql through the command line to the tests. `run-tests localhost user pass schema [port]`.

**Comment by andreas-bok-sociomantic** [comment]
A function exists in https://github.com/sociomantic/libdrizzle-redux/blob/master/tests/unit/common.c#L76 for creating a schema. 
Maybe a function could be added to set up a `test_user` as well, so `db config`  is independent of the `env` variables. 
But could run into errors due `privileges`...or

**Comment by andreas-bok-sociomantic** [comment]
The manual does mention the use of environment variables, user privileges and more 

So I vote for not changing test cases but refer to section **3.2 Running the Test Suite**  in the manual

**Comment by andreas-bok-sociomantic** [comment]
This issue will be resolved with #11

**Comment by andreas-bok-sociomantic** [comment]
> This issue will be resolved with #11
